seata-spring-autoconfigure/seata-spring-autoconfigure-core/src/main/java/org/apache/seata/spring/boot/autoconfigure/provider/SeataFileConfigurationProvider.java

```diff
@@ -0,0 +1,83 @@
+package org.apache.seata.spring.boot.autoconfigure.provider;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.seata.common.util.CollectionUtils;
+import org.apache.seata.config.FileConfiguration;
+import org.apache.seata.config.ConfigurationFactory;
+import org.apache.seata.config.file.FileConfig;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.PropertiesPropertySource;
+
+import static org.apache.seata.common.ConfigurationKeys.FILE_ROOT_PREFIX_CONFIG;
+import static org.apache.seata.common.ConfigurationKeys.FILE_ROOT_PREFIX_REGISTRY;
+import static org.apache.seata.common.ConfigurationKeys.SERVER_PREFIX;
+import static org.apache.seata.common.ConfigurationKeys.STORE_PREFIX;
+import static org.apache.seata.common.ConfigurationKeys.METRICS_PREFIX;
+import static org.apache.seata.common.ConfigurationKeys.TRANSPORT_PREFIX;
+import static org.apache.seata.common.ConfigurationKeys.SEATA_FILE_PREFIX_ROOT_CONFIG;
+
+public class SeataFileConfigurationProvider implements ApplicationContextAware, ApplicationListener<ContextRefreshedEvent> {
```

Comment:
我认为你应该将SeataPropertiesLoader改为给client和server共同使用的方式,而不是重复写一些逻辑
I think you should change the SeataPropertiesLoader to a common way for client and server, rather than rewriting some logic

Reply:
Currently, I’m using reflection in loadSessionAndLockModes() to load sessionMode and lockMode from StoreConfig. Do you think this is a better approach, or do you have other suggestions?

Reply:
Using reflection to determine client-side or server-side configuration is not elegant.